Documentation/check-build.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This was ported from the makefile, it can probably be converted
+# to Python (see e.g. https://stackoverflow.com/a/47657926), that
+# way we might be able to call Sphinx only once and make it quicker
+
+source_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+build_dir="${source_dir}/_build"
+warnings="${build_dir}/warnings.txt"
+spelling="${build_dir}/spelling/output.txt"
+
+target="${1:-"html"}"
+
+cd "${source_dir}"
+
+build_with_spellchecker() {
+    sphinx-build -b spelling -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E 2>&1 > "${warnings}"
+}
+
+build_with_linkchecker() {
+    sphinx-build -b linkcheck -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E
+}
+
+filter_warnings() {
+    grep -v \
```

Comment:
I think that the `grep` match here is not quite having the intended effect, because the return code is based on the matched expressions (even if the output for those matches is excluded), it's not based on whether there is other content in the file.

Maybe for the first run, you could just run this grep then pipe the output to `wc -l` and if there's more than zero lines, we need to print them out and fail out.

See the ignored WARNING below about the `admin_guide` link, the build was skipping past that and only complaining about my spelling:

```
$ git diff                                                                                                                                                                     [0/1878]
diff --git a/Documentation/troubleshooting.rst b/Documentation/troubleshooting.rst
index f2e1d29a9947..4f6c896f068f 100644
--- a/Documentation/troubleshooting.rst
+++ b/Documentation/troubleshooting.rst
@@ -6,6 +6,8 @@

 .. _admin_guide:

+@Im'a shootin' for trouble
+
 ###############
 Troubleshooting
 ###############
$ make -C Documentation/
Sending build context to Docker daemon  10.24kB
Step 1/5 : FROM docker.io/library/python:3.7.5-alpine3.10
 ---> 8922d588eec6
Step 2/5 : LABEL maintainer="maintainer@cilium.io"
 ---> Using cache
 ---> be4f82463a09
Step 3/5 : RUN apk add --no-cache --virtual --update     aspell-en     bash     ca-certificates     enchant     git     py-pip     python     sphinx-python     && true
 ---> Using cache
 ---> 3d659eafb699
Step 4/5 : ADD ./requirements.txt /tmp/requirements.txt
 ---> Using cache
 ---> ef81ad334356
Step 5/5 : RUN pip install -r /tmp/requirements.txt
 ---> Using cache
 ---> 17243537fa83
Successfully built 17243537fa83
Successfully tagged cilium/docs-builder:latest
  GEN   Documentation/_api
  GEN   Documentation/_build/html
Validating documentation (syntax, spelling)...
/usr/local/lib/python3.7/site-packages/sphinx/application.py:402: RemovedInSphinx20Warning: app.info() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)
/src/Documentation/index.rst:32: WARNING: undefined label: admin_guide (if the link has no caption the label must precede a section header)
/usr/local/lib/python3.7/site-packages/sphinx/application.py:388: RemovedInSphinx20Warning: app.warning() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)
WARNING: Found 2 misspelled words
WARNING: Not copying tabs assets! Not compatible with spelling builder

Please fix the following spelling mistakes:
* Documentation/troubleshooting.rst:9: (Im'a)
* Documentation/troubleshooting.rst:9: (shootin)

If the words are not misspelled, run:
Documentation/update-spelling_wordlist.sh Im\\'a shootin
make: *** [Makefile:28: html] Error 1

```

Reply:
For some reason the redirect with `2>&1 > warnings.txt` ceased working actually, but simple `2>` solves the issues.

The output looks like this now:

```
$ make html
tar c requirements.txt Dockerfile \
	  | docker build --tag cilium/docs-builder -
Sending build context to Docker daemon  10.24kB
Step 1/6 : FROM docker.io/library/python:3.7.5-alpine3.10
 ---> 8922d588eec6
Step 2/6 : LABEL maintainer="maintainer@cilium.io"
 ---> Using cache
 ---> f341bfd1da19
Step 3/6 : RUN apk add --no-cache --virtual --update     aspell-en     bash     ca-certificates     enchant     git     py-pip     python     sphinx-python     && true
 ---> Using cache
 ---> f2ee248a51ae
Step 4/6 : ADD ./requirements.txt /tmp/requirements.txt
 ---> Using cache
 ---> 9b21a9a640e6
Step 5/6 : RUN pip install -r /tmp/requirements.txt
 ---> Using cache
 ---> a0be0a621785
Step 6/6 : RUN install -m 0777 -d /usr/local/lib/python3.7/site-packages/versionwarning/_static/data
 ---> Using cache
 ---> 93f6ab089dec
Successfully built 93f6ab089dec
Successfully tagged cilium/docs-builder:latest
cp -r ../api _api
docker container run --rm --workdir /src/Documentation --volume /Users/ilya/Code/cilium/Documentation/..:/src --user "501:20" cilium/docs-builder ./check-build.sh html
Validating documentation (syntax, spelling)...

Please fix the following documentation warnings:
/src/Documentation/index.rst:32: WARNING: undefined label: admin_guide (if the link has no caption the label must precede a section header)

Please fix the following spelling mistakes:
* Documentation/troubleshooting.rst:9: (bdabl) 
* Documentation/troubleshooting.rst:9: (dasd) 

If the words are not misspelled, run:
Documentation/update-spelling_wordlist.sh bdabl dasd 
make: *** [html] Error 1
```